routes/route.js

```diff
@@ -1,101 +1,189 @@
+    );
+  } catch (err) {
+    res.send(err);
+  }
 });
 
-router.post('/login',async(req,res) => {
-   
-    const user =  await Client.findOne({username : req.body.username});
-    if(!user ) { return res.send("Username entered is invalid")}
-    
-    const password = await bcrypt.compare(req.body.password,user.password);
-    if (!password) { return res.send("password entered for the username is incorrect")}
+// route to login user
+router.post('/login', async (req, res) => {
+  try {
+    // to check whether username exists
+    const user = await Client.findOne({ username: req.body.username });
+    if (!user) {
+      return res.send('Username entered is invalid');
+    }
+
+    // to check for correct password
+    const password = await bcrypt.compare(req.body.password, user.password);
+    if (!password) {
+      return res.send('password entered for the username is incorrect');
+    }
 
-    res.status(500).header('Token',user._id).send(`logged -in`)
+    const token = jwt.sign({ _id: user._id }, process.env.SECRET, {
+      expiresIn: 60 * 60,
+    });
+
+    res.status(500).header('Token', token).send(`logged-in`);
+  } catch (err) {
+    res.send(err);
+  }
 });
 
-router.get('/get',async(req,res) => {
-    const token =  req.headers.token
-    if (!token) {return res.send("Login required : Please login before accessing protected routes")} 
+// route to get user details
+router.get('/get', verify, async (req, res) => {
+  try {
+    const user = await Client.findOne({ _id: req.client._id });
+    res.send(user);
+  } catch (err) {
+    res.status(400).send(err);
+  }
+});
 
-    try {
-        const valid = await Client.findOne({_id : token})
-        if (!valid) {return res.send("Login error : please login again")}
-        res.send(valid)
-    } catch (err) {
-        return res.status(400).send("Invalid login")
-    }
-    
-    
+// route to delete request
+router.put('/delete', verify, async (req, res) => {
+  try {
+    await Client.deleteOne({ _id: req.client._id });
+    res.send('User Deleted');
+  } catch (err) {
+    res.send(err);
+  }
 });
 
-router.put('/delete', async(req,res) => {
-    try{
-    const token =  req.headers.token
-    if (!token) {return res.send("Login required : Please login before accessing protected routes")} 
-
-    
-    const valid = await Client.findOne({_id : token})
-    
-    const deleted = await Client.deleteOne({username : valid.username})
-    
-    if (deleted.deleteCount == 1) { return res.send("User Deleted Successfully.")}
-    } catch (err) {
-    res.send("deletion Error")
+// route to get all users
+router.get('/list/:page', async (req, res) => {
+  try {
+    const page = parseInt(req.params.page);
+    if (page < 0) {
+      return res.send('Invalid request');
     }
+
+    const startIndex = (page - 1) * 10;
+    const endIndex = page * 10;
+
+    const data = await Client.find({});
+
+    const result = data.slice(startIndex, endIndex);
```

Comment:
no need to perform slice operation on data .
also finding all the data will decrease the performance (considering large dataset)